### ISSUE TITLE
Stuck with RuntimeError: "Integrity Failure" after 2 hours of working perfectly

### ISSUE THREAD

**Issue by Plasmator-x64** [adafruit/Adafruit_CircuitPython_MPRLS#17]
Stuck with RuntimeError: "Integrity Failure" after 2 hours of working perfectly

File "/usr/local/lib/python3.7/dist-packages/adafruit_mprls.py", line 146, in _read_data
raise RuntimeError("Integrity failure")

**Comment by ladyada** [comment]
is it repeatable?

**Comment by Plasmator-x64** [comment]
Yes - 100% repeatable

Stuck with "Integrity failure" even after RESET. As well, buffer[0] = 0x44 which causes this error.

How do I clear the buffer or reset converter ? Don't see any functions for doing this.

**Comment by ladyada** [comment]
is there something in the datasheet that could do it? if so it can be added

**Comment by Plasmator-x64** [comment]
I don't see anything in the datasheet other than the RESET pin.

The "Integrity failure" refers to a MEMORY or CHECKSUM error.

Now - I have 2 of these Adafruit MPRLS 3965 sensors in a non-functional state. Ugh.

**Comment by ladyada** [comment]
ok - could try to power cycle them

**Comment by Plasmator-x64** [comment]
Yes - tried power cycle , software re-install , re-wire - no change in error

Weird it worked for about 2 hours and then suddenly gets stuck with MEMORY errors.

**Comment by ladyada** [comment]
not sure what it is - never heard of it before :/ you could try opening a ticket with honeywell, only they have access to internal debug info

**Comment by Plasmator-x64** [comment]
Well - I found 2 other mentions (google) of the exact same issue but no solution posted.

At this point - will try RMA back to DigiKey

**Comment by hairem** [comment]
I'm getting this same error and I did a rewire and reset the unit using it pins. Has there been a resolution? According to Honeywell the chip should reset the Memory integrity/error flag upon reboot or power cycle.
